### PR TITLE
Add command line script

### LINKED ISSUE #60: Request to provide command line interface program
Will it be a good idea to introduce a command line program such as those provided in fontTools (eg. ttx, fonttools varlib.mutator), fontmake and AFDKO (eg. tx, sfntedit)? This allow beginners who just want to convert a font from binary to UFO to just install and run.

I have written one such program which might be useful: https://github.com/NightFurySL2001/py-fonttools-script/blob/main/convert_ufo.py

### PR DESCRIPTION
**Pull request by rimas-kudelis**
This rebases and #35 from master and adds some small improvements over it, namely:
- turns `ufoLib2` from hard to an optional dependency
- attempts to fall back to `defcon` if `ufoLib2` is not installed
- outputs status messages when running the script

Fixes #35, fixes #60 (although the script is still not as fancy as the one suggested in #60).

### CODE REVIEW

**Comment by benkiel** [comment]
Can we add an optional command line option to choose between `ufoLib2` and `defcon` for those who have both installed? I know this is a small fussy detail, but it might be helpful for some

**Comment by rimas-kudelis** [comment]
Ugh,  tried to avoid messing with the args parser, but okay, I guess it's time to grow up. I'll try.  :smile:

**Comment by benkiel** [comment]
@rimas-kudelis steal from here: https://github.com/robotools/extractor/pull/35#issuecomment-782049355

**Comment by rimas-kudelis** [comment]
Yep, I'm looking at it, but it won't be just stealing. That command requires that the input font is TTF/OTF, for example, and extractor supports more formats than just that. I'll add the output format argument from that script though.

I'm on it already.

**Comment by rimas-kudelis** [comment]
@benkiel, done. I could still add even more options, like maybe one for the destination directory.

**Comment by rimas-kudelis** [comment]
Ouch, it seems defcon doesn't support the `overwrite` argument to `save()`. How should I handle this? Shall I just remove this option, or require it when `defcon` is to be used as the backend?

File: Lib/extractor/__init__.py [hunk 1]
```diff
@@ -75,3 +75,65 @@ def extractUFO(
...
+    from sys import exit
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser(
+        description="Extract data from font binaries and build UFO objects from them.",
+        epilog="Each resulting UFO will be saved as FONT_FILE.ufo(z) in the same directory as the original FONT_FILE.",
+    )
+    parser.add_argument('FONT_FILE', help='Input font path', nargs="+")
+    parser.add_argument('-m', '--ufo-module', choices=['ufoLib2', 'defcon'], help='Select the default library for writing UFOs (default: autodetect, prefer ufoLib2)')
+    parser.add_argument('-z', '--zip', action="store_true", help="Output UFO ZIP")
```
Comment (benkiel) [inline Lib/extractor/__init__.py:96]: `help="Output UFOZ"`

**Comment by benkiel** [comment]
@rimas-kudelis Looking at the code for overwrite in ufoLib2, I would say we should just remove the overwrite option. If you want to keep it, would say "This will not work if backend is `defcon`" and then do a check for the backend, if it's defcon, do not pass overwrite to save.

File: Lib/extractor/__init__.py [hunk 2]
```diff
@@ -75,3 +75,65 @@ def extractUFO(
...
+
+    structure="zip" if args.zip else "package"
+    had_write_errors = False
+    for font_path in args.FONT_FILE:
+        ufo_path = f"{font_path}.ufo" if not args.zip else f"{font_path}.ufoz"
+        print(f"Extracting {ufo_path}... ", end="")
+        ufo = Font()
+        extractUFO(font_path, ufo)
+        try:
+            ufo.save(ufo_path, overwrite=args.overwrite, structure=structure)
```
Comment (benkiel) [inline Lib/extractor/__init__.py:132]: could just drop overwrite here for simplicity

File: Lib/extractor/__init__.py (hunk 1 above)
Comment (rimas-kudelis) [reply Lib/extractor/__init__.py:96]: Hm, "UFO ZIP" is actually how the specification calls it. :)

Comment (benkiel) [reply Lib/extractor/__init__.py:96]: Cool. I can't remember, and I am used to thinking UFOZ, but fine with UFO ZIP

Comment (rimas-kudelis) [reply Lib/extractor/__init__.py:96]: From https://unifiedfontobject.org/versions/ufo3/#storage-formats:
> UFO has two possible storage formats. The first, “UFO Package”, is a multi-file, file system directory structure. The second, “UFO ZIP”, is a [ZIP archive](https://en.wikipedia.org/wiki/Zip_(file_format)) containing the same directory structure as defined for UFO Package.

I'll leave this as-is then.

File: Lib/extractor/__init__.py (hunk 2 above)
Comment (rimas-kudelis) [reply Lib/extractor/__init__.py]: Okay, I've removed the option, and added an explicit check for file/dir existence. If it exists, that input file will not be converted.
I reckon that's better than having two different scenarios, right? 
Hm, maybe I should make a PR for defcon adding this option though... :laughing:

**Review by benkiel — APPROVED** [review]

**Comment by benkiel** [comment]
Thank you! Going to push this out as a new release

**Comment by rimas-kudelis** [comment]
I think maybe we should update the readme first.

**Comment by NightFurySL2001** [comment]
Thanks @rimas-kudelis for the awesome job! The parent folder part in my script is just to organise the files easier, in the end I end up not using that option at all 😂

**Comment by rimas-kudelis** [comment]
@NightFurySL2001, thank you for your script as well! As you see, I took some inspiration from it. :)

BTW I had some doubts about whether I should accept an unlimited list of input files or instead accept only one input file and allow specifying output file name like you did, but I guess I was too lazy to change that bit. But I do think that maybe your approach is better, since looping could have easily be done externally in the shell. Oh well... :)

**Pull request merged**